src/components/ContentEditable.native.tsx

```diff
@@ -43,18 +44,33 @@ const ContentEditable = ({
   forceUpdate,
   placeholder,
   isEditing,
+  isTable,
   ...props
 }: ContentEditableProps) => {
   const allowInnerHTMLChange = useRef<boolean>(true)
   const { width } = useDimensions().window
   const [height, setHeight] = useState(DEFAULT_WEBVIEW_HEIGHT)
   const contentRef = useRef<WebView>(null)
+  const [innerHTMLValue, setInnerHTMLValue] = useState(html)
+
+  const [webviewHTML, setWebviewHTML] = useState<string>(
+    createWebHTML({ innerHTML: html, placeholder, isEditing, isTable }),
+  )
 
-  const [webviewHTML, setWebviewHTML] = useState<string>(createWebHTML({ innerHTML: html, placeholder, isEditing }))
   useEffect(() => {
-    setWebviewHTML(createWebHTML({ innerHTML: html, placeholder, isEditing }))
+    setWebviewHTML(createWebHTML({ innerHTML: html, placeholder, isEditing, isTable }))
+  }, [isTable])
+
+  useEffect(() => {
+    setWebviewHTML(createWebHTML({ innerHTML: html, placeholder, isEditing, isTable }))
   }, [isEditing])
 
+  useEffect(() => {
+    if (html.trim() === innerHTMLValue.trim()) return
+
+    setWebviewHTML(createWebHTML({ innerHTML: html, placeholder, isEditing, isTable }))
+  }, [html])
+
```

Comment:
You can use a single `useEffect` with `isTable`, `isEditing` and `html` as dependencies.

Reply:
I split into two different effects, `isEditing` and `isTable` are very similar so they are in a single `useEffect`. Whereas `html` effect is independent so it can perform the better.